README.md

```diff
@@ -224,9 +225,59 @@ Some more notes:
  * See `bin/wasm-opt --help` for the full list of options and passes.
  * Passing `--debug` will emit some debugging info.
 
+### wasm2js
+
+Run
+
+```
+bin/wasm2js [input.wasm file]
+```
+
+This will print out JavaScript to the console.
+
+For example, try
+
+```
+$ bin/wasm2js test/hello_world.wast
+```
+
+That output contains
+
+```
+ function add(x, y) {
+  x = x | 0;
+  y = y | 0;
+  return x + y | 0 | 0;
+ }
+```
+
+as a translation of
+
+```
+ (func $add (; 0 ;) (type $0) (param $x i32) (param $y i32) (result i32)
+  (i32.add
+   (local.get $x)
+   (local.get $y)
+  )
+ )
+```
+
+You can also tell wasm2js to optimize, using the normal optimization flags
+wasm-opt and other tools receive (such as `-Os`). For optimal code size,
+you should both optimize and run a JavaScript minifier afterwards.
+
+Things to keep in mind with wasm2js's output:
+
+ * It is not possible to match WebAssemblty semantics 100% precisely with fast
+   JavaScript code. For example, every load and store may trap, and to make
+   JavaScript do the same we'd need to add checks everywhere, which would be
+   large and slow. Instead, wasm2js assumes loads and stores do not trap, that
+   int/float conversions do not trap, and so forth. There may also be slight
```

Comment:
Probably worth replacing "and so forth" with a definitive list at some point.

Reply:
Yeah, the list of wasm2js differences is still being worked out, but we should write them out when we stabilize.